src/SFML/Audio/AudioDevice.cpp

```diff
@@ -25,211 +25,348 @@
-AudioDevice::AudioDevice()
+AudioDevice::AudioDevice() : m_impl(std::make_unique<Impl>())
 {
-    // Create the device
-    audioDevice = alcOpenDevice(nullptr);
+    // Create the log
+    m_impl->m_log.emplace();
 
-    if (audioDevice)
+    if (auto result = ma_log_init(nullptr, &*m_impl->m_log); result != MA_SUCCESS)
     {
-        // Create the context
-        audioContext = alcCreateContext(audioDevice, nullptr);
+        m_impl->m_log.reset();
+        err() << "Failed to initialize the audio log: " << ma_result_description(result) << std::endl;
+        return;
+    }
 
-        if (audioContext)
-        {
-            // Set the context as the current one (we'll only need one)
-            alcMakeContextCurrent(audioContext);
-
-            // Apply the listener properties the user might have set
-            float orientation[] = {listenerDirection.x,
-                                   listenerDirection.y,
-                                   listenerDirection.z,
-                                   listenerUpVector.x,
-                                   listenerUpVector.y,
-                                   listenerUpVector.z};
-            alCheck(alListenerf(AL_GAIN, listenerVolume * 0.01f));
-            alCheck(alListener3f(AL_POSITION, listenerPosition.x, listenerPosition.y, listenerPosition.z));
-            alCheck(alListenerfv(AL_ORIENTATION, orientation));
-        }
-        else
+    // Register our logging callback to output any warning/error messages
+    if (auto result = ma_log_register_callback(&*m_impl->m_log,
+                                               ma_log_callback_init(
+                                                   [](void*, ma_uint32 level, const char* message)
+                                                   {
+                                                       if (level <= MA_LOG_LEVEL_WARNING)
+                                                           err() << "miniaudio " << ma_log_level_to_string(level)
+                                                                 << ": " << message << std::flush;
+                                                   },
+                                                   nullptr));
+        result != MA_SUCCESS)
+        err() << "Failed to register audio log callback: " << ma_result_description(result) << std::endl;
+
+    // Create the context
+    m_impl->m_context.emplace();
+
+    auto contextConfig = ma_context_config_init();
+    contextConfig.pLog = &*m_impl->m_log;
+
+    if (auto result = ma_context_init(nullptr, 0, &contextConfig, &*m_impl->m_context); result != MA_SUCCESS)
+    {
+        m_impl->m_context.reset();
+        err() << "Failed to initialize the audio context: " << ma_result_description(result) << std::endl;
+        return;
+    }
+
+    // Create the playback device
+    m_impl->m_playbackDevice.emplace();
+
+    auto playbackDeviceConfig         = ma_device_config_init(ma_device_type_playback);
```

Comment:
The `sound` example is very noisy on my machine, but it sounds good if I set `playbackDeviceConfig.playback.format = ma_format_f32;` here.

Reply:
The miniaudio implementation seems to be a bit bugged when using any other device format but `ma_format_f32`. The format is forced to `ma_format_f32` in the latest commit.